custom_components/pentair_cloud/coordinator.py

```
@@ -52,19 +52,68 @@ def get_devices(self, device_type: str | None = None) -> list[dict]:
+                            # Normalize the fields to match expected format
+                            if "fields" in device_data:
+                                device_data["fields"] = self._normalize_fields(
+                                    device_data["fields"]
+                                )
+                            # Merge detailed data INTO original device to preserve
+                            # fields like lastReport that only exist in get_devices()
+                            merged_device = {**device, **device_data}
+                            detailed_devices.append(merged_device)
+                        else:
+                            detailed_devices.append(device)
+                    except Exception:  # pylint: disable=broad-except
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)